### PR TITLE
Nc news 4 api articles articleid

### PR DESCRIPTION
**Pull request by KashaM767**

### CODE REVIEW

File: __tests__/seed.test.js [hunk 1]
```diff
@@ -39,4 +39,32 @@ describe('GET /api/topics', () => {
     })
 });
 
+describe('GET /api/articles/:article_id', () => {
+    test('GET:200 sends a single article to the client', () => {
+        return request(app)
+            .get('/api/articles/1')
+            .expect(200)
+            .then(({ body }) => {
```
Comment (nc-jonathan-mcguire) [inline __tests__/seed.test.js:60]: Your responses need to be returned with a key i.e. `article` in this case

File: __tests__/seed.test.js [hunk 2]
```diff
@@ -39,4 +39,32 @@ describe('GET /api/topics', () => {
...
+    });
+    test("404: responds with an error message if the article_id does not exist", () => {
+        return request(app)
+            .get('/api/articles/26')
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("not found")
+            })
+    });
+});
```
Comment (nc-jonathan-mcguire) [inline __tests__/seed.test.js:88]: We need a 400 test for a bad id.

File: controllers/controllers.js [hunk 3]
```diff
@@ -8,5 +8,11 @@ exports.getTopics = (req, res, next) => {
         .catch(next)
 };
 
+exports.getArticleById = (req, res, next) => {
+    const { article_id } = req.params
+    retrieveArticleById(article_id).then((rows) => {
+        res.status(200).send(rows)
```
Comment (nc-jonathan-mcguire) [inline controllers/controllers.js]: Your responses need to be sent back in an object with a key - i this case {article:rows}

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
Just two bits to add please:

- [ ] Send back responses in objects with keys
- [ ] Missing a test: Status 400: Bad request, invalid article_id
E.g. GET /api/articles/notanumber

**Comment by nc-mick-fay** [comment]
Hi Kasha, happy to approve this pull request now these changes have been made. Merge into main when you're ready

**Pull request merged**